From f8a2bd16f61b2eaf4a3c4220ffccefcd4ff14d53 Mon Sep 17 00:00:00 2001
From: Benjamin Berg <benjamin.berg@intel.com>
Date: Fri, 27 Sep 2024 11:36:22 +0200
Subject: [PATCH] tests: Fix rare EAPOL frame ID wrap-around failure

In rare cases ieee8021x_set_conf would fail as the ID of the EAPOL frame
wrapped around and the frame was not ignored as expected.

Signed-off-by: Benjamin Berg <benjamin.berg@intel.com>
---
 tests/hwsim/test_ieee8021x.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/tests/hwsim/test_ieee8021x.py b/tests/hwsim/test_ieee8021x.py
index db0b69576..7453aecba 100644
--- a/tests/hwsim/test_ieee8021x.py
+++ b/tests/hwsim/test_ieee8021x.py
@@ -440,7 +440,10 @@ def test_ieee8021x_set_conf(dev, apdev):
     dev[0].connect("ieee8021x-open", key_mgmt="IEEE8021X", eapol_flags="0",
                    eap="PSK", identity="psk.user@example.com",
                    password_hex="0123456789abcdef0123456789abcdef",
-                   scan_freq="2412")
+                   scan_freq="2412",
+                   # The workaround in eap_success_workaround can trigger a
+                   # rare failure when the EAPOL frame ID wraps.
+                   eap_workaround="0")
 
     addr0 = dev[0].own_addr()
     tests = ["EAPOL_SET 1",
-- 
2.43.0

